### PR TITLE
Ockam command transports API

### PR DESCRIPTION
**Pull request by spacekookie**
Also add a `transport` command to the ockam command to create and
manage transports for nodes.

## Checks

- [ ] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits).
- [ ] All commits in this Pull Request follow the Ockam [commit message convention](https://www.ockam.io/learn/how-to-guides/contributing/CONTRIBUTING#commit-messages).
- [ ] I accept the Ockam Community [Code of Conduct](https://www.ockam.io/learn/how-to-guides/high-performance-team/conduct/).
- [ ] I have accepted the Ockam [Contributor License Agreement](https://www.ockam.io/learn/how-to-guides/contributing/cla/) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/contributors](https://github.com/build-trust/contributors) repository.

### CODE REVIEW

File: implementations/rust/ockam/ockam_api/Cargo.toml [hunk 1]
```diff
@@ -25,6 +25,7 @@ lmdb          = ["std", "lmdb-rkv"]
 vault-storage = ["ockam_vault/storage"]
 
 [dependencies]
+ockam       = { path = "../ockam", version = "^0.60.0", features = ["software_vault"] }
```
Comment (twittner) [inline implementations/rust/ockam/ockam_api/Cargo.toml:28]:
```suggestion
ockam       = { path = "../ockam", version = "0.60.0", features = ["software_vault"] }
```

File: implementations/rust/ockam/ockam_api/src/lib.rs [hunk 2]
```diff
@@ -11,8 +11,12 @@ use minicbor::encode::{self, Encoder, Write};
 use minicbor::{Decode, Encode};
 use ockam_core::compat::borrow::Cow;
 use ockam_core::compat::rand;
+use std::fmt::{Display, Formatter};
```
Comment (twittner) [inline implementations/rust/ockam/ockam_api/src/lib.rs]:
```suggestion
use core::fmt::{Display, Formatter};
```

File: implementations/rust/ockam/ockam_api/src/lib.rs [hunk 3]
```diff
@@ -89,6 +93,22 @@ pub enum Method {
...
+            f,
+            "{}",
+            match self {
+                Self::Get => "GET",
+                Self::Post => "POST",
+                Self::Put => "PUT",
+                Self::Delete => "DELETE",
+                Self::Patch => "PATCH",
+            }
+        )
```
Comment (twittner) [inline implementations/rust/ockam/ockam_api/src/lib.rs]:
```suggestion
        f.write_str(
            match self {
                Self::Get => "GET",
                Self::Post => "POST",
                Self::Put => "PUT",
                Self::Delete => "DELETE",
                Self::Patch => "PATCH",
            }
        )
```

File: implementations/rust/ockam/ockam_api/src/nodes/types.rs [hunk 4]
```diff
@@ -1,106 +1,156 @@
...
-#[derive(Debug, Clone, Encode, Decode)]
+/// Encode which type of transport is being requested
+#[derive(Copy, Clone, Debug, Decode, Encode)]
 #[rustfmt::skip]
-#[cbor(map)]
-pub struct CreateNode<'a> {
-    #[cfg(feature = "tag")]
-    #[n(0)] tag: TypeTag<4925167>,
-    #[b(1)] name: Cow<'a, str>
+pub enum TransportType {
```
Comment (twittner) [inline implementations/rust/ockam/ockam_api/src/nodes/types.rs]: Just wanted to mention that if an enum will never carry data in its variants, one may consider using [`#[cbor(index_only)]`](https://twittner.gitlab.io/minicbor/minicbor_derive/index.html#cborindex_only).

File: implementations/rust/ockam/ockam_api/src/nodes/types.rs [hunk 5]
```diff
@@ -1,106 +1,156 @@
...
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "{}",
+            match self {
+                Self::Tcp => "TCP",
+                Self::Ble => "BLE",
+                Self::WebSocket => "Websocket",
+            }
+        )
```
Comment (twittner) [inline implementations/rust/ockam/ockam_api/src/nodes/types.rs]:
```suggestion
        f.write_str(
            match self {
                Self::Tcp => "TCP",
                Self::Ble => "BLE",
                Self::WebSocket => "Websocket",
            }
        )
```

File: implementations/rust/ockam/ockam_api/src/nodes/types.rs [hunk 6]
```diff
@@ -1,106 +1,156 @@
...
+impl Display for TransportMode {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "{}",
+            match self {
+                Self::Listen => "Listening",
+                Self::Connect => "Remote connection",
+            }
+        )
```
Comment (twittner) [inline implementations/rust/ockam/ockam_api/src/nodes/types.rs]:
```suggestion
        f.write_str(
            match self {
                Self::Listen => "Listening",
                Self::Connect => "Remote connection",
            }
        )
```

File: implementations/rust/ockam/ockam_api/src/nodes/types.rs [hunk 7]
```diff
@@ -1,106 +1,156 @@
...
-    pub fn new<S: Into<Cow<'a, str>>>(name: S) -> Self {
-        CreateNode {
+impl<'a> NodeStatus<'a> {
+    pub fn new(node_name: &'a str, status: &str, workers: u32, pid: i32, transports: u32) -> Self {
+        Self {
             #[cfg(feature = "tag")]
             tag: TypeTag,
-            name: name.into(),
+            node_name: Cow::Borrowed(node_name),
+            status: Cow::Owned(status.into()),
```
Comment (twittner) [inline implementations/rust/ockam/ockam_api/src/nodes/types.rs]: Any reason why status has to be owned here?

File: implementations/rust/ockam/ockam_api/src/nodes/types.rs [hunk 8]
```diff
@@ -1,106 +1,156 @@
...
-    pub fn status(&self) -> &str {
-        &self.status
+impl<'a> TransportStatus<'a> {
+    pub fn new(tt: TransportType, tm: TransportMode, addr: &String) -> Self {
+        Self {
+            #[cfg(feature = "tag")]
+            tag: TypeTag,
+            tt,
+            tm,
+            addr: addr.clone().into(),
```
Comment (twittner) [inline implementations/rust/ockam/ockam_api/src/nodes/types.rs]:
```suggestion
    pub fn new<S: Into<Cow<'a, str>>>(tt: TransportType, tm: TransportMode, addr: S) -> Self {
        Self {
            #[cfg(feature = "tag")]
            tag: TypeTag,
            tt,
            tm,
            addr: addr.into(),
```

File: implementations/rust/ockam/ockam_api/Cargo.toml (hunk 1 above)
Comment (mrinalwadhwa) [reply implementations/rust/ockam/ockam_api/Cargo.toml:28]: Our release scripts manage/update versions at release time within our crates @metaclips may have a recommendation based on what the scripts expect.

**Review by mrinalwadhwa — APPROVED** [review]

**Pull request merged**